python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -5008,12 +5008,14 @@ def test_cov_nans():
         operator.truediv,
         operator.mod,
         operator.pow,
-        operator.eq,
-        operator.lt,
-        operator.le,
-        operator.gt,
-        operator.ge,
-        operator.ne,
+        # comparison ops will temporarily XFAIL
```

Comment:
So, this PR broke tests for series and dataframe mixed arithmetic, because of course comparison ops are now producing different results. However upon investigating further, I realized that dataframe/series mixed arithmetic actually isn't working at all in some situations, and this test has a grievous error (of my own making of course) where it doesn't actually compare the results properly at the end of the test, hence the failure going unnoticed. 

This led to PR https://github.com/rapidsai/cudf/pull/7491. However, these PRs are now mutually dependent in the sense that until this PR is merged, I can't actually fix that functionality unless I spend time making the results match the logic this PR is about to change. 

It doesn't make sense to me to fix broken code to produce results that will shortly thereafter be changed, forcing me to redo the code/test again. 

Let me know if this seems like a sensible way to proceed.

Reply:
cc @cwharris @shwina

Reply:
Synced offline and this plan seems like the least painful path forward (without having to merge 3 different PRs into a single one)